client/src/holdings/AddTradeDialog.tsx

```diff
@@ -1,22 +1,25 @@
 import './AddTradeDialog.css';
 
-import {Field, Form, Formik} from 'formik';
-import {TextField} from 'formik-material-ui';
-import {KeyboardDateTimePicker} from 'formik-material-ui-pickers';
 import React, {useState} from 'react';
+import {Controller, useForm} from 'react-hook-form';
 
 import DateFnsUtils from '@date-io/date-fns';
+import {yupResolver} from '@hookform/resolvers/yup';
+import {Select} from '@material-ui/core';
 import Button from '@material-ui/core/Button';
-import CircularProgress from '@material-ui/core/CircularProgress';
 import Dialog from '@material-ui/core/Dialog';
 import DialogActions from '@material-ui/core/DialogActions';
 import DialogContent from '@material-ui/core/DialogContent';
 import DialogTitle from '@material-ui/core/DialogTitle';
-import {MuiPickersUtilsProvider} from '@material-ui/pickers';
+import TextField from '@material-ui/core/TextField';
+import ToggleButton from '@material-ui/lab/ToggleButton';
+import ToggleButtonGroup from '@material-ui/lab/ToggleButtonGroup';
+import {DateTimePicker, MuiPickersUtilsProvider} from '@material-ui/pickers';
 
 import {API} from '../api';
 import {CURRENCIES} from '../constants';
 import {CreateTradeData, Holding, TradeCategory} from '../types';
+import {tradeValidationSchema} from './tradevalidationschema';
```

Comment:
This file doesn't seem to be included in your PR, but it should be called tradeValidationSchema. React uses UpperCamelCase for component files and lowerCamelCase for other ones.